src/directory_listing/directory_info.rs

```diff
@@ -33,9 +33,19 @@ impl DirectoryInfo {
         })
     }
 
-    /// Get the unique ID representing this directory in the network
-    pub fn get_key(&self) -> (&::routing::NameType, u64) {
-        (&self.id, self.type_tag)
+    /// Returns the key representing the directory listing
+    pub fn get_key(&self) -> (&::routing::NameType, u64, bool, &::AccessLevel) {
```

Comment:
The return type has become very complicated. It is better to define a struct like:
```
pub struct DirectoryKey<'a> {
    id                 : &'a ::routing::NameType,
    tag_type      : u64,
    is_versioned: bool,
    access_level: &'a ::AccessLevel,
}

impl DirectoryKey<`a> {
    pub fn new( ......) { ..... }
    pub fn get_id(&self) -> &::routing::NameType { ..... }
    .....
}
```
Put this in a `pub` module of-course as other crates will need to access it too.